src/NuGet.Clients/NuGet.SolutionRestoreManager/RestoreManagerPackage.cs

```diff
@@ -72,6 +78,13 @@ public sealed class RestoreManagerPackage : AsyncPackage
                 _buildEvents = dte.Events.BuildEvents;
                 _buildEvents.OnBuildBegin += BuildEvents_OnBuildBegin;
 
+                var solutionBuildManager = (IVsSolutionBuildManager5)await GetServiceAsync(typeof(SVsSolutionBuildManager));
+                if (solutionBuildManager != null)
```

Comment:
Is it possible for this to be null? It might be better to fail if we can't hook up this event. We wouldn't know that future hangs aren't due to this getting missed.